module-2/lab-hypothesis-testing/your-code/challenge-1.ipynb

```diff
@@ -19,13 +19,14 @@
   },
```

Comment:
- A pesar de que pueda resultar confuso el lab te envío algunos comentarios en este challenge. En este challenge, debemos hacer un planteamiento inicial sobre qué tipo de test vamos a hacer. Podemos considerar nuestro dataset de pokemon como una única población de la que vamos a sacar diferentes muestras sobre las que contrastaremos si las medias muestrales son significativamente iguales. Si partimos de este supuesto, dos muestras de nuestra población deberíamos correr un Paired test lo que en Python se traduce en la función: 
```Python 
scipy.stats.ttest_rel(a, b, axis=0)
```
- Otra forma de chequear single type vs dual type using loc:
```Python
single_type = pokemon.loc[pokemon['Type 2'].isnull()]
dual_type = pokemon.loc[pokemon['Type 2'].notnull()]
t_test_features(single_type, dual_type)
```
- Cuando python nos muestra notación científica puede llevarnos a confusión, una forma de evitarlo es convertir a float el resultado para verlo completo. 
```Python
format(valor, 'f')
format(valor, '.<número de decimales>f')
```
- "Now, we want to compare whether there are significant differences of Attack vs Defense and  Sp. Atk vs Sp. Def of all pokemons. Please write your code below.Hint: are you comparing different populations or the same population?" Esta pregunta dependiendo de como se arme el argumentario podríamos considerarlo también paired test, es decir, distintas muestras de una misma población. Si consideras, contraste de medias de muestras independientes de diferentes poblaciones el método de scipy es scipy.stats.ttest_ind(a, b, axis=0, equal_var=True) en lugar de ttest_rel.